scipy/stats/tests/test_distributions.py

```diff
@@ -3067,64 +3100,121 @@ def test_pdf_nolan_samples(self):
                 1,0 # gamma, delta
                 2 # output file
         """
-        fn = os.path.abspath(os.path.join(os.path.dirname(__file__),
-                                          'data/stable-pdf-sample-data.npy'))
-        data = np.load(fn)
+        data = np.load(
+            Path(__file__).parent /
+            'data/levy_stable/stable-Z1-pdf-sample-data.npy'
+        )
 
-        data = np.core.records.fromarrays(data.T, names='x,p,alpha,beta')
+        data = np.core.records.fromarrays(data.T, names='x,p,alpha,beta,pct')
 
         # support numpy 1.8.2 for travis
         npisin = np.isin if hasattr(np, "isin") else np.in1d
 
+        # some tests break on linux 32 bit
+        uname = platform.uname()
+        is_linux_32 = uname.system == 'Linux' and uname.machine == 'i686'
+        platform_desc = "/".join(
+            [uname.system, uname.machine, uname.processor])
+
+        # fmt: off
         tests = [
-            # best selects
-            ['best', None, 8, None],
-
-            # quadrature is accurate for most alpha except 0.25; perhaps
-            # limitation of Nolan stablec?
-            # we reduce size of x to speed up computation as numerical
-            # integration slow.
-            ['quadrature', None, 8,
-             lambda r: ((r['alpha'] > 0.25) &
-                        (npisin(r['x'], [-10, -5, 0, 5, 10])))],
-
-            # zolatarev is accurate except at alpha==1, beta != 0
-            ['zolotarev', None, 8, lambda r: r['alpha'] != 1],
```

Comment:
Another place to use pytest.mark.parameterize. See other review comment above.

Reply:
I can see the advantage to using pytest's parameterize decorator especially in your other comments. I don't feel it adds much benefit to the pdf/cdf testing. These tests don't fail on individual test points and it was very useful having all the test data in one array for checking general hot spots when developing this PR (and the original PR). I feel using pytest parameterize here will further complicate these tests.

Reply:
OK, after mulling this over, I don't think you need have to have every case as a separate pytest param for the pdf/cdf tests either. It seems like you've already taken care to produce informative messages when the tests fail.

A better thing to do would be to still use pytest.parameterize, but have three parameters each containing an entire array of test cases. The first would have only a small number of test cases, just enough to hit each execution path at least once. These would be for regular tests that are expected to run in under one second. The next array of test cases would be marked slow and contain a fairly representative sample with size chosen so that each test will run in under 10 seconds when slow tests are run. The third array would contain an exhaustive sample of test cases, be marked xslow. As it stands, at least on my machine, the tests run too long even for ones marked slow. Also, it would be good to have some regular tests too for quick checking.